### ISSUE TITLE
Server crashes/restarts in a loop

### ISSUE THREAD

**Issue by thomze** [StrangeLoopGames/EcoIssues#4473]
Suddenly the server started to restart constantly. 
Server is running again, with the use of a backup file. 

This was the error from the dump file:

```
> --BEGIN DUMP--
> Dump Time
> 02/13/2018 22:08:41
> 
> Exception
> System.InvalidOperationException: Changes could not be applied, the requested operation would result in an invalid inventory state.
>    at Eco.Gameplay.Items.InventoryChangeSet.ApplyInternal(Boolean throwOnInvalid)
>    at Eco.Gameplay.Items.Inventory.Modify(Action`1 callback, User user)
>    at Eco.Gameplay.Components.StockpileComponent.TryUpdateInventoryFromWorld()
>    at System.Collections.Generic.List`1.ForEach(Action`1 action)
>    at Eco.Gameplay.Objects.WorldObject.Tick()
>    at Eco.Shared.Utils.EnumerableExtensions.ForEach[T](IEnumerable`1 enumeration, Action`1 action)
>    at Eco.Gameplay.Objects.WorldObjectManager.Run()
>    at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)
>    at System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)
>    at System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state)
>    at System.Threading.ThreadHelper.ThreadStart(Object obj)
> 
> --END DUMP--
```

**Comment by Aragas** [comment]
Used dnSpy and replaced Apply() with ApplyInternal(false);
Server seems to be working fine for now. It's understandable this will show problems in long run, but either this or losing the world.

**Comment by MarjieAVolk** [comment]
@alexandrage shared this save file where they had the same error: https://cloud.mail.ru/public/4qDa/Dx9shnwa8

**Comment by gcoupelant** [comment]
We've had the same problem on our server. We found that some stockpiles were duplicated in place (from previous crashes).
When we removed the stockpile, a new one filled with stuff took its place.
One of our stockpile had to be empty 6 time before all "instances" were in our backpack 😱

**Comment by metachronism** [comment]
fixed for 7.2, will report an error, but not crash out the server while we track it down more
